### PR TITLE
Fix/connected calendar component

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Adds react/testing-library to package.json
- Modifies jest.config.ts file in order to be able to run test on components.
- Creates a new component to display when No calendar is connected to user's account.
- Implements single render test for component.
- Fixes DestinationCalendarSelector component when externalId string its too long.

I was using SSR to pass a prop to the view wether NoCalendarConnectedAlert should display, opted to load calendars via TRCP

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Try to uninstall any connected calendar and then go to /event-types, alert should be showing.
- [ ] Try to connect a calendar return to page /event-types and alert should be gone.

### CODE REVIEW

File: apps/web/components/DestinationCalendarSelector.tsx [hunk 1]
```diff
@@ -70,7 +70,13 @@ const DestinationCalendarSelector = ({
...
         })),
     })) ?? [];
-
+  const defaultCalendarSelectedString = destinationCalendar?.externalId
+    ? `(${
+        destinationCalendar.externalId.length > 15
+          ? destinationCalendar.externalId.substring(0, 15) + "..."
+          : destinationCalendar.externalId
+      })`
+    : "";
```
Comment (alannnc) [inline apps/web/components/DestinationCalendarSelector.tsx:79]: shortening long strings for calendar integrations

File: apps/web/components/ui/LinkText.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import Link from "next/link";
+
+interface ILinkTextProps {
+  href: string;
+  children: React.ReactNode;
+  classNameChildren?: string;
+}
+// This component had to be made in order to make i18n work with next/link
+// Reference: https://github.com/i18next/react-i18next/issues/1090#issuecomment-615426145
```
Comment (alannnc) [inline apps/web/components/ui/LinkText.tsx:9]: ^ comment

File: apps/web/components/ui/NoCalendarConnectedAlert/__tests__/NoCalendarConnectedAlert.test.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  describe("Render test", () => {
+    it("should render without crashing", () => {
+      // Disabled as its asking for full trpc useQuery response
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      jest.spyOn(trpc, "useQuery").mockReturnValue({
+        isSuccess: true,
+        isFetched: true,
+        data: { connectedCalendars: [], destinationCalendar: undefined },
+      });
```
Comment (alannnc) [inline apps/web/components/ui/NoCalendarConnectedAlert/__tests__/NoCalendarConnectedAlert.test.tsx:17]: With this we can mock our desired trpc useQuery responses and components should comply.

File: apps/web/components/ui/NoCalendarConnectedAlert/index.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { useTranslation } from "next-i18next";
+import { Trans } from "next-i18next";
+
+import { trpc } from "@calcom/trpc/react";
+// @TODO: alert v2
```
Comment (alannnc) [inline apps/web/components/ui/NoCalendarConnectedAlert/index.tsx:5]: Not sure if we should already pull v2 for Alert component
@sean-brydon

File: apps/web/components/ui/NoCalendarConnectedAlert/index.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import { Alert } from "@calcom/ui/Alert";
+
+import { LinkText } from "../LinkText";
+
+const NoCalendarConnectedAlert = () => {
+  const { t } = useTranslation();
+
+  const query = trpc.useQuery(["viewer.connectedCalendars"]);
+  // We are not gonna show this alert till we fetch data from DB
+  let defaultCalendarConnected = true;
```
Comment (alannnc) [inline apps/web/components/ui/NoCalendarConnectedAlert/index.tsx:15]: Defaults to true, so we don't display it till we load DB data.

File: apps/web/jest.config.ts [hunk 6]
```diff
@@ -7,7 +7,7 @@ const config: Config.InitialOptions = {
   verbose: true,
   roots: ["<rootDir>"],
   setupFiles: ["<rootDir>/test/jest-setup.js"],
-  testMatch: ["**/test/lib/**/*.(spec|test).(ts|tsx|js)"],
+  testMatch: ["**/test/lib/**/*.(spec|test).(ts|tsx|js)", "**/__tests__/**/*.(spec|test).(ts|tsx|js)"],
```
Comment (alannnc) [inline apps/web/jest.config.ts:10]: This adds functionality so when we create test next to our components we can use this folder structure.
- components/
   - myComponent/
       - index.tsx
       - __test__/
          - myComponent.test.ts

File: apps/web/pages/event-types/index.tsx [hunk 7]
```diff
@@ -59,17 +56,7 @@ interface EventTypeListProps {
...
-  group,
-  readOnly,
-  connectedCalendars,
-}: {
-  type: EventType;
-  group: EventTypeGroup;
-  readOnly: boolean;
-  connectedCalendars: ConnectedCalendars[] | undefined;
-}) => {
+const Item = ({ type, group, readOnly }: { type: EventType; group: EventTypeGroup; readOnly: boolean }) => {
```
Comment (alannnc) [inline apps/web/pages/event-types/index.tsx:59]: This was added on a previous PR, but its no longer needed.

File: apps/web/pages/event-types/index.tsx [hunk 8]
```diff
@@ -597,31 +576,17 @@ const EventTypesPage = (props: InferGetServerSidePropsType<typeof getServerSideP
                   message={
                     <Trans i18nKey="plan_upgrade_instructions">
                       You can
-                      <a href="/api/upgrade" className="underline">
+                      <LinkText href="/api/upgrade" classNameChildren="underline">
```
Comment (alannnc) [inline apps/web/pages/event-types/index.tsx:579]: Updating to LinkText component that let us use <Link> inside <Trans>

File: apps/web/pages/event-types/index.tsx [hunk 9]
```diff
@@ -652,26 +617,4 @@ const EventTypesPage = (props: InferGetServerSidePropsType<typeof getServerSideP
   );
 };
 
-export async function getServerSideProps(context: GetServerSidePropsContext) {
```
Comment (alannnc) [inline apps/web/pages/event-types/index.tsx:655]: No longer required, opted to use TRPC on client.

File: apps/web/jest.config.ts (hunk 6 above)
Comment (zomars) [reply apps/web/jest.config.ts:10]: How about components that don't have this structure right now?

Can we use:
- MyComponent.tsx
- MyComponent.test.tsx

in the same level?

Comment (zomars) [inline apps/web/jest.config.ts:10]:
```suggestion
  testMatch: ["**/*.(spec|test).(ts|tsx|js)"],
```

Let's allow test to coexist with the rest of the code WDYT?

File: apps/web/components/ui/LinkText.tsx [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
```
Comment (zomars) [inline apps/web/components/ui/LinkText.tsx:10]:
```suggestion
/**
 * This component had to be made in order to make i18n work with next/link
 * @see https://github.com/i18next/react-i18next/issues/1090#issuecomment-615426145
 **/
```

Modded comment so it actually displays on hover:
image

Comment (alannnc) [reply apps/web/components/ui/LinkText.tsx]: Awesome

File: apps/web/jest.config.ts (hunk 6 above)
Comment (alannnc) [reply apps/web/jest.config.ts:10]: I would go for "**/__tests__/**/*.(spec|test).(ts|tsx|js)" and then renaming test/lib/** to __test__/lib

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**